tests/drivers/can/api/src/canfd.c

```diff
@@ -445,6 +605,100 @@ ZTEST_USER(canfd, test_set_timing_data_max)
+	zassert_equal(err, 0, "failed to stop CAN controller (err %d)", err);
+
+	test_timing = *max_timing;
+	test_timing.sjw += 1;
+	err = can_set_timing_data(can_dev, &test_timing);
+	zassert_equal(err, -ENOTSUP, "invalid sjw was accepted (err %d)", err);
+
+	test_timing = *max_timing;
+	test_timing.prop_seg += 1;
+	err = can_set_timing_data(can_dev, &test_timing);
+	zassert_equal(err, -ENOTSUP, "invalid prop_seg was accepted (err %d)", err);
+
+	test_timing = *max_timing;
+	test_timing.phase_seg1 += 1;
+	err = can_set_timing_data(can_dev, &test_timing);
+	zassert_equal(err, -ENOTSUP, "invalid phase_seg1 was accepted (err %d)", err);
+
+	test_timing = *max_timing;
+	test_timing.phase_seg2 += 1;
+	err = can_set_timing_data(can_dev, &test_timing);
+	zassert_equal(err, -ENOTSUP, "invalid phase_seg2 was accepted (err %d)", err);
+
+	test_timing = *max_timing;
+	test_timing.prescaler += 1;
+	err = can_set_timing_data(can_dev, &test_timing);
+	zassert_equal(err, -ENOTSUP, "invalid prescaler was accepted (err %d)", err);
+
+	err = can_start(can_dev);
+	zassert_equal(err, 0, "failed to start CAN controller (err %d)", err);
+}
+
+/**
+ * @brief Test that timing values for the data phase that are below the minimum can't be set.
+ */
+ZTEST_USER(canfd, test_set_timing_data_below_min)
+{
+	int err;
+
+	const struct can_timing * min_timing = can_get_timing_data_min(can_dev);
+	struct can_timing test_timing;
+
+	err = can_stop(can_dev);
+	zassert_equal(err, 0, "failed to stop CAN controller (err %d)", err);
+
+	test_timing = *min_timing;
+	test_timing.sjw -= 1;
+	err = can_set_timing_data(can_dev, &test_timing);
+	zassert_equal(err, -ENOTSUP, "invalid sjw was accepted (err %d)", err);
+
+	test_timing = *min_timing;
+	test_timing.sjw -= min_timing->phase_seg1 + 1;
+	err = can_set_timing_data(can_dev, &test_timing);
+	zassert_equal(err, -ENOTSUP, "invalid sjw was accepted (err %d)", err);
+
+	test_timing = *min_timing;
+	test_timing.sjw -= min_timing->phase_seg2 + 1;
+	err = can_set_timing_data(can_dev, &test_timing);
+	zassert_equal(err, -ENOTSUP, "invalid sjw was accepted (err %d)", err);
+
+	test_timing = *min_timing;
+	test_timing.prop_seg -= 1;
+	err = can_set_timing_data(can_dev, &test_timing);
+	zassert_equal(err, -ENOTSUP, "invalid prop_seg was accepted (err %d)", err);
+
+	test_timing = *min_timing;
+	test_timing.phase_seg1 -= 1;
+	err = can_set_timing_data(can_dev, &test_timing);
+	zassert_equal(err, -ENOTSUP, "invalid phase_seg1 was accepted (err %d)", err);
+
+	test_timing = *min_timing;
+	test_timing.phase_seg2 -= 1;
+	err = can_set_timing_data(can_dev, &test_timing);
+	zassert_equal(err, -ENOTSUP, "invalid phase_seg2 was accepted (err %d)", err);
+
+	test_timing = *min_timing;
+	test_timing.prescaler -= 1;
+	err = can_set_timing_data(can_dev, &test_timing);
+	zassert_equal(err, -ENOTSUP, "invalid prescaler was accepted (err %d)", err);
+
+	err = can_start(can_dev);
+	zassert_equal(err, 0, "failed to start CAN controller (err %d)", err);
+}
```

Comment:
Same here.

Reply:
All fields in can_timing structure are unsigned.
https://docs.zephyrproject.org/latest/doxygen/html/structcan__timing.html
I added skip when value is 0.